module/VuFindSearch/src/VuFindSearch/Backend/Blender/Backend.php

```diff
@@ -0,0 +1,590 @@
+
+            $offsetOk = $backendOffsets[$backendAtPos]
+                < $backendTotals[$backendAtPos];
+            $record = $offsetOk ? $this->getRecord(
+                $backendDetails[$backendAtPos],
+                $backendRecords[$backendAtPos],
+                $backendOffsets[$backendAtPos]++,
+                $blockSize
+            ) : null;
+
+            if (null === $record) {
+                // Try other backends:
+                foreach ($availableBackendIds as $backendId) {
+                    if ($backendId === $backendAtPos) {
+                        continue;
+                    }
+                    $offsetOk = $backendOffsets[$backendId]
+                        < $backendTotals[$backendId];
+                    $record = $offsetOk ? $this->getRecord(
+                        $backendDetails[$backendId],
+                        $backendRecords[$backendId],
+                        $backendOffsets[$backendId]++,
+                        $blockSize
+                    ) : null;
+
+                    if (null !== $record) {
+                        break;
+                    }
+                }
+            }
+
+            if (null === $record) {
+                break;
+            }
+            $mergedCollection->add($record, false);
+        }
+    }
+
+    /**
+     * Retrieve a single document.
+     *
+     * @param string   $id     Document identifier
+     * @param ParamBag $params Search backend parameters
+     *
+     * @return \VuFindSearch\Response\RecordCollectionInterface
+     */
+    public function retrieve($id, ParamBag $params = null)
+    {
+        throw new \Exception('Blender does not support retrieve');
+    }
+
+    /**
+     * Return the record collection factory.
+     *
+     * Lazy loads a generic collection factory.
+     *
+     * @return RecordCollectionFactoryInterface
+     */
+    public function getRecordCollectionFactory()
+    {
+        return null;
+    }
+
+    /**
+     * Get active backends for a search
+     *
+     * @param ParamBag $params Search backend parameters
+     *
+     * @return array
+     */
+    protected function getActiveBackends(ParamBag $params): array
+    {
+        $activeBackends = $this->backends;
+
+        // Handle the blender_backend pseudo-filter
+        $fq = $params->get('fq') ?? [];
+        $filteredBackends = [];
+        foreach ($fq as $filter) {
+            $advanced = preg_match(
+                '/\{!tag=blender_backend_filter}blender_backend:\((.+)\)/',
+                $filter,
+                $matches
+            );
+            if ($advanced) {
+                $filter = explode(' OR ', $matches[1]);
+            }
+            foreach ((array)$filter as $current) {
+                if (strncmp($current, 'blender_backend:', 16) === 0) {
+                    $active = trim(substr($current, 16), '"');
+                    if (!isset($activeBackends[$active])) {
+                        throw new \Exception(
```

Comment:
Is this a fatal error or a warning? If we use OR filters and provide a set of values of which three are valid and one is not, do we really want that to blow everything up? (Maybe we do since it indicates that somebody is tampering with something... but maybe it's better to just log a warning, in case it's related to a recent configuration transition or something like that).

Reply:
Good point. I changed that to log a warning and updated the test accordingly.